src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -1178,14 +1178,32 @@ void CascadiaSettings::_validateThemeExists()
         auto newTheme = winrt::make_self<Theme>();
         newTheme->Name(L"system");
         _globals->AddTheme(*newTheme);
-        _globals->Theme(L"system");
+        _globals->Theme(*winrt::make_self<ThemePair>(L"system"));
     }
 
-    if (!_globals->Themes().HasKey(_globals->Theme()))
+    if (_globals->Theme().DarkName() == _globals->Theme().LightName())

```

Comment:
BTW I think it'd be advantageous for us in the long term if we consistently stopped calling WinRT functions repeatedly like that. These calls are about 4x as expensive as regular, non-inlined function calls and returning a runtimeclass reference involves at least 2 such calls each time. I realize that even then these calls are plenty fast for "cold paths" like this, but I think that it might lead to impacts down the road, for instance by impacting Debug build performance to a point where it begins to be annoying (Debug build WinRT calls are over 20x more expensive), or just because one of these function calls ends up not being `O(1)` anymore but e.g. `O(log N)` or something (an ordered map lookup instead of an unordered one for instance).

Just to be clear, I'm not saying that this is a problem right now, but I'm hoping to make a case for what I think might be beneficial for our long term code health.